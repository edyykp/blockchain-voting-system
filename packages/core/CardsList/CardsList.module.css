.container {
  display: flex;
  flex-wrap: wrap;
  padding: 5rem;
  justify-content: space-around;
  gap: 4rem;
}

@media screen and (max-width: 1337px) {
  .container {
    gap: 3rem;
  }
}

@media screen and (max-width: 1305px) {
  .container {
    padding: 3rem;
    padding-bottom: 5rem;
  }
}

@media screen and (max-width: 1241px) {
  .container {
    padding: 2rem;
    padding-bottom: 5rem;
  }
}

@media screen and (max-width: 1209px) {
  .container {
    gap: 1rem;
    padding: 0.5rem;
    padding-bottom: 5rem;
  }
}
