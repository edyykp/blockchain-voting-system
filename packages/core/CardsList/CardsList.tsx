import styles from './CardsList.module.css';

import { RaceType } from '@packages/types';
import { Card } from '@packages/components';

type CardsListProps = {
  races: RaceType[];
};

export const CardsList = ({ races }: CardsListProps) => {
  return (
    <div className={styles.container} data-testid="cards-list">
      {races.map((race, key) => (
        <Card race={race} key={key} />
      ))}
    </div>
  );
};
