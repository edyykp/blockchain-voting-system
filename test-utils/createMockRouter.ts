import { NextRouter } from 'next/router';

export function createMockRouter(router: Partial<NextRouter>): NextRouter {
  return {
    basePath: '',
    pathname: '/',
    route: '/',
    asPath: '/',
    query: {},
    isFallback: false,
    back: jest.fn(),
    beforePopState: jest.fn(),
    prefetch: jest.fn(),
    push: jest.fn(),
    reload: jest.fn(),
    replace: jest.fn(),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isLocaleDomain: false,
    isReady: false,
    defaultLocale: 'en',
    domainLocales: [],
    isPreview: false,
    ...router,
  };
}
