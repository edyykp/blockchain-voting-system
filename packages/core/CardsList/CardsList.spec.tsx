import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { RouterContext } from 'next/dist/shared/lib/router-context';
import { createMockRouter } from 'test-utils/createMockRouter';

import { CardsList } from './CardsList';

const mockRaces = {
  races: [
    {
      circuitId: 'imola',
      locality: 'Imola',
      country: 'Italy',
    },
    {
      circuitId: 'miami',
      locality: 'Miami',
      country: 'USA',
    },
    {
      circuitId: 'monaco',
      locality: 'Monaco',
      country: 'Monaco',
    },
  ],
};

const setup = () => {
  return {
    ...render(
      <RouterContext.Provider value={createMockRouter({})}>
        <CardsList {...mockRaces} />
      </RouterContext.Provider>,
    ),
  };
};

describe('CardsListâ†’', () => {
  it('renders', () => {
    const { container, getByTestId } = setup();

    expect(getByTestId('cards-list')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
