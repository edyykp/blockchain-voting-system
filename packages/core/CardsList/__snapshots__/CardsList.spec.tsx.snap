// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsListâ†’ renders 1`] = `
<div>
  <div
    class="container"
    data-testid="cards-list"
  >
    <div
      class="container"
    >
      <div
        class="imgWrap"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="imola"
            class="image"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="body"
      >
        <h1
          class="locality"
        >
          Imola
          ,
        </h1>
        <h2
          class="country"
        >
          Italy
        </h2>
        <button
          class="button votingButton"
        >
          Vote now
        </button>
        <button
          class="button standingsButton"
        >
          Standings
        </button>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="imgWrap"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="miami"
            class="image"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="body"
      >
        <h1
          class="locality"
        >
          Miami
          ,
        </h1>
        <h2
          class="country"
        >
          USA
        </h2>
        <button
          class="button votingButton"
        >
          Vote now
        </button>
        <button
          class="button standingsButton"
        >
          Standings
        </button>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="imgWrap"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="monaco"
            class="image"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="body"
      >
        <h1
          class="locality"
        >
          Monaco
          ,
        </h1>
        <h2
          class="country"
        >
          Monaco
        </h2>
        <button
          class="button votingButton"
        >
          Vote now
        </button>
        <button
          class="button standingsButton"
        >
          Standings
        </button>
      </div>
    </div>
  </div>
</div>
`;
